backup/restore

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env bash
+
+## Restore from a backup
+set -eE
+
+volumes=("hint_db_data" "hint_redis_data" "hint_prerun" "hint_results" "hint_uploads" "hint_config")
+
+array_contains() {
+    local seeking=$1
+    shift
+    local in=1
+    for element; do
+        if [[ $element == "$seeking" ]]; then
+            in=0
+            break
+        fi
+    done
+    return $in
+}
+
+assert_file_exists() {
+    if !([ ! -z "$1" ] && [ -f "$1" ]); then
+        echo "Backup dir $1 does not exist"
+        echo "Usage: restore <backup_dir>"
+        exit 1
+    fi
+}
+
+assert_all_data_exists() {
+    if !([ ! -z "$1" ] && [ -d "$1" ] && [ -f "$1/db_dump.sql" ] && [ -f "$1/redis_backup.tar" ] &&
+        [ -f "$1/prerun_backup.tar" ] && [ -f "$1/results_backup.tar" ] && [ -f "$1/uploads_backup.tar" ]); then
+        echo "Backup dir $1 is incomplete"
+        echo "Usage: restore <backup_dir>"
+        exit 1
+    fi
+}
+
+create_volumes() {
+    existing_volumes=($(docker volume ls --format '{{.Name}}'))
+    for volume in ${volumes[@]}; do
+        if array_contains "$volume" "${existing_volumes[@]}"; then
+            echo "Volume '${volume}' already exists, remove volume before running restore"
+            exit 1
+        fi
+    done
+    for volume in ${volumes[@]}; do
+        echo "Creating volume ${volume}"
+        docker volume create ${volume}
+    done
+}
+
+restore() {
+    create_volumes
+    docker run --rm -d -v hint_db_data:/pgdata -e POSTGRES_PASSWORD=password -e PGDATA=/pgdata --name restore_pg postgres:10.3
+    sleep 5 ## Ideally here we would do something cleverer and poll for db being ready to accept connections
```

Comment:
See https://github.com/vimc/orderly/blob/master/docker/schema/postgres_wait for an example, if you want one. Here it would require you to copy it into the container, then run it I think. 5s is probably ok given it's an empty db though

Reply:
Thanks, added that proper wait in now too